internal/kubeclient/transport.go

```diff
@@ -6,27 +6,33 @@ import (
 	netutil "k8s.io/apimachinery/pkg/util/net"
 )
 
+const (
+	xKargoUserCredentialHeader = "X-Kargo-User-Credential"
+)
+
 var (
-	_ netutil.RoundTripperWrapper = &authRoundTripper{}
+	_ netutil.RoundTripperWrapper = &credentialHook{}
```

Comment:
What effect does this line have? It appears you're allocating a `credentialHook`, but not doing anything with it.

Reply:
This makes sure that `&credentialHook{}` implements `netutil.RoundTripperWrapper` interface: ([Verify Interface Compliance](https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance) from uber-go styleguide)

Reply:
Neat trick.